oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneDocumentMaker.java

```diff
@@ -315,6 +312,29 @@ protected boolean indexTypeOrderedFields(Document doc, String pname, int tag, Pr
         }
         return fieldAdded;
     }
+    
+    private static BytesRef checkTruncateLength(String prop, String value, String path, int maxLength) {
```

Comment:
This is not always working properly. E.g. this throws a StringIndexOutOfBoundsException:

    checkTruncateLength("x", "$$$$€£$$", "/x", 10)